File: ChatAppServer/Schema/Queries/AuthQuery.cs
using ChatAppServer.Repositories;
using ChatAppServer.Repositories.IRepositories;
using ChatAppServer.Schema.Types;
using ChatAppServer.Schema.Types.IType;
using ChatAppServer.Services;
using ChatAppServer.Services.IServices;
using System.Security.Claims;

namespace ChatAppServer.Schema.Queries
{
    [ExtendObjectType(typeof(Query))]
    public class AuthQuery
    {
        public async Task<ResultType> GetRefreshRoken(string email, [Service]IAuthRepository userRepository, [Service] IJwtService jwtService)
        {
            var user = await userRepository.GetUserByEmail(email);
            if(user == null)
            {
                return new ResultType(true, ["User not found"], null);
            }

            //var t = new IResultType();

            return new ResultType(true, [], new UserType
            {
                Email = user.Email!,
                PhoneNumber = user.PhoneNumber,
                FullName = user.FullName,
                AvatarUrl = user.AvatarUrl,
                Token = await jwtService.CreateJwt(user),
                UniqueCodeUser = user.UniqueCodeUser,
            });
        }
    }
}

File: ChatAppClient/src/app/graphql/queries/authQuery.ts
import { gql } from 'apollo-angular';

export const REFRESH_TOKEN = gql`
	query ($email: String!) {
		refreshRoken(email: $email) {
			data {
				... on UserType {
					token
				}
			}
		}
	}
`;

export const LOGIN = gql`
	query ($email: String!, $password: String!) {
		login(model: { email: $email, password: $password }) {
			success
			errorMessages
			data {
				... on UserType {
					email
					fullName
					avatarUrl
					phoneNumber
					token
					uniqueCodeUser
				}
			}
		}
	}
`;
